plutus-core/plutus-core/src/PlutusCore/Default/Universe.hs

```diff
@@ -233,96 +234,108 @@ instance Pretty (DefaultUni a) where
 instance Pretty (SomeTypeIn DefaultUni) where
     pretty (SomeTypeIn uni) = pretty uni
 
-instance DefaultUni `Contains` Integer                    where knownUni = DefaultUniInteger
-instance DefaultUni `Contains` BS.ByteString              where knownUni = DefaultUniByteString
-instance DefaultUni `Contains` Text.Text                  where knownUni = DefaultUniString
-instance DefaultUni `Contains` ()                         where knownUni = DefaultUniUnit
-instance DefaultUni `Contains` Bool                       where knownUni = DefaultUniBool
-instance DefaultUni `Contains` []                         where knownUni = DefaultUniProtoList
-instance DefaultUni `Contains` (,)                        where knownUni = DefaultUniProtoPair
-instance DefaultUni `Contains` Data                       where knownUni = DefaultUniData
-instance DefaultUni `Contains` BLS12_381.G1.Element       where knownUni = DefaultUniBLS12_381_G1_Element
-instance DefaultUni `Contains` BLS12_381.G2.Element       where knownUni = DefaultUniBLS12_381_G2_Element
-instance DefaultUni `Contains` BLS12_381.Pairing.MlResult where knownUni = DefaultUniBLS12_381_MlResult
+-- | Elaborate a built-in type (see 'ElaborateBuiltin') from 'DefaultUni'.
+type ElaborateBuiltinDefaultUni :: forall a. a -> a
+type family ElaborateBuiltinDefaultUni x where
+    ElaborateBuiltinDefaultUni (f x) = ElaborateBuiltinDefaultUni f `TyAppRep` x
+    ElaborateBuiltinDefaultUni x     = BuiltinHead x
+
+type instance ElaborateBuiltin DefaultUni x = ElaborateBuiltinDefaultUni x
 
 instance (DefaultUni `Contains` f, DefaultUni `Contains` a) => DefaultUni `Contains` f a where
     knownUni = knownUni `DefaultUniApply` knownUni
 
-instance KnownBuiltinTypeAst DefaultUni Integer                    => KnownTypeAst DefaultUni Integer
-instance KnownBuiltinTypeAst DefaultUni BS.ByteString              => KnownTypeAst DefaultUni BS.ByteString
-instance KnownBuiltinTypeAst DefaultUni Text.Text                  => KnownTypeAst DefaultUni Text.Text
-instance KnownBuiltinTypeAst DefaultUni ()                         => KnownTypeAst DefaultUni ()
-instance KnownBuiltinTypeAst DefaultUni Bool                       => KnownTypeAst DefaultUni Bool
-instance KnownBuiltinTypeAst DefaultUni [a]                        => KnownTypeAst DefaultUni [a]
-instance KnownBuiltinTypeAst DefaultUni (a, b)                     => KnownTypeAst DefaultUni (a, b)
-instance KnownBuiltinTypeAst DefaultUni Data                       => KnownTypeAst DefaultUni Data
-instance KnownBuiltinTypeAst DefaultUni BLS12_381.G1.Element       => KnownTypeAst DefaultUni BLS12_381.G1.Element
-instance KnownBuiltinTypeAst DefaultUni BLS12_381.G2.Element       => KnownTypeAst DefaultUni BLS12_381.G2.Element
-instance KnownBuiltinTypeAst DefaultUni BLS12_381.Pairing.MlResult => KnownTypeAst DefaultUni BLS12_381.Pairing.MlResult
-
-{- Note [Constraints of ReadKnownIn and MakeKnownIn instances]
```

Comment:
The Note is gone, because now the instances are all defined the same straightforward way regardless of whether the built-in type is monomorphic or polymorphic.